### ISSUE TITLE
Rolling dimension order for individual image layers instead of all layers

### ISSUE THREAD

**Issue by rbavery** [napari/napari#991]
I have two image layers, one originates from importing multiple .tif files at once into an Image layer, each is ordered (3035, 2016, 3). Another is a single .tif file with shape 
(3035, 2016, 52) imported individually.

![Selection_090](https://user-images.githubusercontent.com/22258697/75100178-5e541e00-557f-11ea-9979-e3b67a720725.png)

When I select the option to roll the dimensions, I can only see one dataset at a time, I think because the other image's dimensions get rolled as well. clicking the roll dimensions button will correctly display one layer but display the other image layer with the wrong dimension order.

The alternative is not bad, I used `np.moveaxis` to reorder the (3035, 2016, 52) sized tif to (52, 3035, 2016), but it might be nice to have this option built in to napari. Thanks for making this, big fan of this tool!

**Comment by sofroniewn** [comment]
That's great that you're liking the tool @rbavery! From the sounds of it you only really want to do this roll once when you load in the image write? If that's the case, it might be something we expose in a more advanced file open dialog we're working on in https://github.com/napari/napari/pull/616#issuecomment-562896633 or once we get #937 merged it will be possible for you to write your own IO plugin that does the rolling if you are often working with data like this.

How do these options sound? Or do you really want to select a layer and then have the roll button apply only to the selected layers - which is possible, but we'd need to make sure that wasn't confusing in the UI or there was a clear what to turn that "only-selected" functionality on or off

**Comment by rbavery** [comment]
Yep, that's correct. Maybe this "only-selected" option for dimension rolling could go up near the other options that seem to apply to selected image layers (interpolation, blending, etc.). If writing an IO plugin was simple enough I'd make one that automatically moved the axis with the smallest length to the first position and that would work too (for .tifs with 3 axis since this is usually what I'd like to do). In any case, I'll start to make a habit of saving my .tifs in image-channel-first order for when I want to load them in napari.

**Comment by sofroniewn** [comment]
Ok sounds good @rbavery - maybe we leave this one to sit a little as we think about exactly how that would work on the UI and as we work to make the io-plugins possible. Thanks for bringing this to our attention though

**Comment by liaprins-czi** [comment]
I am just checking in about this issue, to see if anything has changed since it was created, and if we'd want to consider updating anything in the UI to accommodate a solution ... or is there an io plugin that canhelp, or @rbavery are you content with your afformentioned plan to make a "habit of saving my .tifs in image-channel-first order for when I want to load them in napari"?

If we do want to solve this from within napari viewer, I can create a design issue to look into it from design perspective. Thanks!
